src/QsCompiler/CompilationManager/EditorSupport/SymbolInfo.cs

```diff
@@ -277,13 +342,168 @@ bool OverlapsWithPosition(Range symRange)
+                QsExpressionKind.OR or => Binary(or.Item1, or.Item2),
+                QsExpressionKind.BOR bOr => Binary(bOr.Item1, bOr.Item2),
+                QsExpressionKind.BAND bAnd => Binary(bAnd.Item1, bAnd.Item2),
+                QsExpressionKind.BXOR bXor => Binary(bXor.Item1, bXor.Item2),
+                QsExpressionKind.LSHIFT lShift => Binary(lShift.Item1, lShift.Item2),
+                QsExpressionKind.RSHIFT rShift => Binary(rShift.Item1, rShift.Item2),
+                QsExpressionKind.CONDITIONAL cond => Many(new[] { cond.Item1, cond.Item2, cond.Item3 }),
+                QsExpressionKind.CopyAndUpdate update => Binary(update.Item1, update.Item3),
+                QsExpressionKind.UnwrapApplication unwrap => DeclarationsInExpression(unwrap.Item, position),
+                QsExpressionKind.AdjointApplication adj => DeclarationsInExpression(adj.Item, position),
+                QsExpressionKind.ControlledApplication ctl => DeclarationsInExpression(ctl.Item, position),
+                QsExpressionKind.CallLikeExpression call => Binary(call.Item1, call.Item2),
+                QsExpressionKind.SizedArray sizedArray => Binary(sizedArray.value, sizedArray.size),
+                QsExpressionKind.Lambda lambda => Lambda(lambda.Item),
+                _ => Enumerable.Empty<LocalVariableDeclaration<string>>(),
+            };
+
+            IEnumerable<LocalVariableDeclaration<string>> Binary(TypedExpression e1, TypedExpression e2) =>
+                DeclarationsInExpression(e1, position).Concat(DeclarationsInExpression(e2, position));
+
+            IEnumerable<LocalVariableDeclaration<string>> Many(IEnumerable<TypedExpression> es) =>
+                es.SelectMany(e => DeclarationsInExpression(e, position));
+
+            IEnumerable<LocalVariableDeclaration<string>> Lambda(Lambda<TypedExpression> lambda)
+            {
+                var inputType = CallableInputType(expression.ResolvedType);
+                var inferred = new InferredExpressionInformation(false, true);
+                return DeclarationsInTypedSymbol(lambda.Param, inputType, inferred)
+                    .Concat(DeclarationsInExpression(lambda.Body, position));
+            }
+
+            static ResolvedType CallableInputType(ResolvedType type) => type.Resolution switch
+            {
+                QsTypeKind.Function function => function.Item1,
+                QsTypeKind.Operation operation => operation.Item1.Item1,
+                _ => throw new Exception("Type is not a callable type."),
+            };
+        }
+
+        private static IEnumerable<LocalVariableDeclaration<string>> DeclarationsInTypedSymbol(
+            QsSymbol symbol, ResolvedType type, InferredExpressionInformation inferred)
+        {
+            switch (symbol.Symbol, type.Resolution)
+            {
+                case (QsSymbolKind.Symbol name, _):
+                    var range = symbol.Range.ValueOr(Range.Zero);
```

Comment:
Hm, I don't see how this would result in correct behavior is the range is not set. Is this ever supposed to be Null? If not, then in the spirit of failing early, it would make more sense to throw. If there is reason to expect that the range is null, why is the declaration return with incorrect information, rather than ignored?

Reply:
Throwing feels too strict because there could be other symbols with ranges in a tuple; if it returned an empty enumerable rather than throwing, it could extract as much useful information as it could. If LocalVariableDeclaration had an optional range, I would just return one without a range if the source symbol was missing a range, but it's required.

Reply:
I guess you can argue that the function can't do what it's claiming to do (return LocalVariableDeclarations for every symbol) if any of the symbols is missing a range, because LocalVariableDeclaration requires a range, so throwing makes sense.

Reply:
Updated code to throw if the symbol range is null.